pkg/provider/handlers/deploy.go

```diff
@@ -83,17 +83,9 @@ func deploy(ctx context.Context, req types.FunctionDeployment, client *container
 
 	name := req.Service
 
-	container, err := client.NewContainer(
-		ctx,
-		name,
-		containerd.WithImage(image),
-		containerd.WithSnapshotter(snapshotter),
-		containerd.WithNewSnapshot(req.Service+"-snapshot", image),
-		containerd.WithNewSpec(oci.WithImageConfig(image),
-			oci.WithCapabilities([]string{"CAP_NET_RAW"}),
-			oci.WithMounts(mounts),
-			oci.WithEnv(envs)),
-	)
+	caps := []string{"CAP_NET_RAW"}
+
+	container, err := service.NewContainer(ctx, *client, name, image, snapshotter, caps, mounts, nil, envs)
```

Comment:
I preferred the old version of this code, what's the reason for the helper method? What does the helper add ?

Reply:
Yes, I agree it's confusing.
Reverting back to call the NewContainer directly instead of the helper.